### DISCUSSION TITLE
Force center-of-mass to body origin?

### DISCUSSION THREAD

**Discussion by DocAce** [jrouwe/JoltPhysics#176] [Q&A] [answered]
In the game I'm working on (a space sim) the artists define an object's center of mass and export the assets so that it's in the coordinate origin. We're using compound shapes for most objects and I'm finding that Jolt always adds an automatically calculated offset to it. This causes some issues, especially with rotational movement. Is there some way I could disable this or is that only achievable with a DecoratedShape that forces the center of mass offset to zero?

**Comment by jrouwe** [comment, marked as answer]
Jolt indeed automatically calculates the center of mass for any created shape. If you want to force the center of mass to the origin you can use the OffsetCenterOfMassShape to do so, you'd create one with: 

```my_new_shape = new OffsetCenterOfMassShape(-my_original_shape->GetCenterOfMass(), my_original_shape)```

**Reply by DocAce** [reply]
Thanks. I think I will instead write my own DecoratedShape so that I don't have to update it every time I'm changing the Compound (which is a Mutable one).

**Comment by jrouwe** [comment]
If you're working with a MutableCompound shape you could also just create an empty one and then Add/Remove Shapes afterwards without calling AdjustCenterOfMass. That will also keep the COM at the origin.

**Reply by DocAce** [reply]
Well, I feel silly now 😄 
I've been doing exactly this all along and don't know how I got it in my head that the shape was calling AdjustCenterOfMass internally...
